packages/fxa-auth-server/lib/routes/recovery-key.js

```diff
@@ -249,6 +260,66 @@ module.exports = (log, db, Password, verifierVersion, customs, mailer) => {
         return db.recoveryKeyExists(uid);
       },
     },
+    {
+      method: 'GET',
+      path: '/recoveryKey/recoveryKeyHint',
+      options: {
+        ...RECOVERY_KEY_DOCS.RECOVERYKEY_HINT_GET,
+        // No auth needed to fetch the recovery key hint
+        auth: false,
+      },
+      handler: async function (request) {
+        log.begin('getRecoveryKeyHint', request);
+
+        const { uid } = request.auth.credentials;
```

Comment:
I've added rate limiting and passed in the uid as query param instead of credentials objection. Passing the uid in the payload was causing remote test failures. @vbudhram could you have another look to see if the changes look good?